### PR TITLE
docs: rewrite

### PR DESCRIPTION
**Pull request by caasion**
I have started a rewrite of the README.md file to hopefully make the plugin easier to understand. I would like to request for you help in writing this because I am not exactly sure what certain features do (the auto-formatting in particular.

I have given you access to the forked repository, so you could directly push any changes there.

I really hope this can attract more people to use your amazing plugin!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,59 +1,118 @@
...
+## Selection Rules
+
+The syntax of a selection rule is as follows:
+
+```
+'<trigger (one char)>' -> '<left insert>' + '<right insert>'
+```
+*Anything in angled brackets are replaced*
+
+When text is selected and **one** character is entered, Typing Transformer will help you insert two characters on each side of the selection.
```
Comment (aptend) [inline README.md:90]: I pushed commit just now, there is no "two characters" limitation any more for `<left insert>` and `<right insert>` 

it is kind of bug 😹

File: README.md [hunk 2]
```diff
@@ -1,59 +1,118 @@
...
+A conversion rule has the following syntax:
 
 ```
-'dpx|' -> 'don\'t panic|'
+'<trigger>' -> '<result>'
 ```
+*Anything in angled brackets are replaced*
+
+### Example 1: Expanding Abbreviated Phrases
+![](https://github.com/caasion/typing-transformer-obsidian/blob/docs/docs/dpx.gif)
```
[the remark this answers is not in the document]
Comment (aptend) [reply README.md]: done

**Comment by caasion** [comment]
Sorry for the delay! I finally had time to finish (somewhat) the docs. But there are somethings that I need your help to do:

More explanation on the auto-formatting and the zone indicator. As I don't really use those features myself, I don't really know much about them but I think it would help if you could answer:

What does it do? Does it only add spaces automatically? Are there other features?
When does it add spaces? What are the criteria for it to automatically add spaces?
What languages does it support? Is it only Chinese and English right now?

Secondly, I would need your help (or anyone's) for the README-CN.md because I am not proficient with simplified Chinese. I am reluctant to use translation tools because I know they won't be 100% accurate. Also, to prevent clutter, I think you could move the 
README-CN.md into the docs. Just make sure to update the link to it in the English README.

**Comment by aptend** [comment]
> What does it do? Does it only add spaces automatically? Are there other features?

It only add spaces.

> When does it add spaces? What are the criteria for it to automatically add spaces?

I wrote some explanation in: https://github.com/aptend/typing-transformer-obsidian/blob/55d7e9372a9cfb7c985049715fda098a9113fd50/How%20it%20works.md?plain=1#L1-L24

A example (i will add it to the doc):
```
Input: 秦时moon汉时关，  万里长征人no还

Block types: 
- Line
  - Cn: "秦时"
     ---- add a space
  - Eng: "moon"
     ---- add a space
  - Cn: "汉时关"
  - FWPunct: "，"
  - MultiSpace: "  "
  - Cn: "万里长征人"
    ----- add a space
  - Eng: "no"
    ----- add a space
  - Cn: "还"
  - EOI: ""
``` 

> What languages does it support? Is it only Chinese and English right now?

yes, it serves mainly for Chineses and English for now

**Comment by aptend** [comment]
> Secondly, I would need your help (or anyone's) for the README-CN.md because I am not proficient with simplified Chinese. I am reluctant to use translation tools because I know they won't be 100% accurate. Also, to prevent clutter, I think you could move the
README-CN.md into the docs. Just make sure to update the link to it in the English README.

Never mind, I will take care of README-CN.md later. Thanks for your wonderful work on README !!

**Comment by caasion** [comment]
Nope, I don't think there will be any updates as of now. It is ready! :confetti_ball:

**Pull request merged**